### DISCUSSION TITLE
Stable IP addresses range for Actions

### DISCUSSION THREAD

**Discussion by fisherking** [community/community#26441] [Actions]
<p>It’s really important to have a list of IP addresses of the build machines for security reasons.<br>
Is it possible to get it with github API? I don’t see this information in <a href="https://api.github.com/meta" rel="nofollow noopener">https://api.github.com/meta</a></p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi,</p>
<p>Windows and Ubuntu hosted runners are hosted in Azure and have the same IP address ranges as Azure Data centers. Currently, all hosted runners are in the East US 2 Azure region, but more regions may added over time.</p>
<p>Microsoft updates the Azure IP address ranges weekly in a JSON file that you can download from the <a href="https://www.microsoft.com/en-us/download/details.aspx?id=56519" rel="nofollow noopener">Azure IP Ranges and Service Tags - Public Cloud</a> website. You can use this range of IP addresses if you require an allow-list to prevent unauthorized access to your internal resources.</p>
<p>Official link:</p>
<p><a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#ip-addresses-of-runners-on-github-hosted-machines" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#ip-addresses-of-runners-on-github-hosted-machines</a></p>

**Comment by aerogus** [comment]
<p>Hi,</p>
<p>are we sure the ip ranges from this list is always up to date ?</p>
<p>I describe my problem here:</p>
<p><a href="https://github.community/t/github-actions-ip-ranges/16824">https://github.community/t5/GitHub-Actions/Github-actions-IP-ranges/m-p/37491</a></p>
<p>thanks for your help</p>

**Comment by haythem** [comment]
<p>Hello,</p>
<p>I’ve created an action that queries the runner’s public IP Address.</p>
<p><a href="https://github.com/haythem/public-ip" rel="nofollow noopener">https://github.com/haythem/public-ip</a></p>

**Comment by dandv** [comment]
<p>This thread is a dupe.</p>
<p>The currently active thread is <a href="https://github.community/t/whitelist-github-workflow/16077">https://github.community/t5/GitHub-Actions/Whitelist-Github-Workflow/td-p/15916</a></p>
